src/hooks/useOperatorTrades.ts

```diff
@@ -71,33 +71,40 @@ export function useOrderTrades(order: Order | null): Result {
   // contrary to useOrder hook, where it searches all networks for a given orderId
   const networkId = useNetworkId()
 
-  const fetchTrades = useCallback(async (networkId: Network, order: Order): Promise<void> => {
-    setIsLoading(true)
-
-    const { uid: orderId, buyToken, sellToken } = order
-
-    try {
-      const trades = await getTrades({ networkId, orderId })
-
-      setTrades(trades.map((trade) => ({ ...transformTrade(trade), buyToken, sellToken })))
-      setError(undefined)
-    } catch (e) {
-      const msg = `Failed to fetch trades`
-      console.error(msg, e)
-      setError({ message: msg, type: 'error' })
-    } finally {
-      setIsLoading(false)
-    }
-  }, [])
+  const fetchTrades = useCallback(
+    async (controller: AbortController, _networkId: Network, _order: Order): Promise<void> => {
+      setIsLoading(true)
+
+      const { uid: orderId, buyToken, sellToken } = _order
+
+      try {
+        const _trades = await getTrades({ networkId: _networkId, orderId })
+
+        if (controller.signal.aborted) return
+
+        setTrades(_trades.map((trade) => ({ ...transformTrade(trade), buyToken, sellToken })))
+        setError(undefined)
+      } catch (e) {
+        const msg = `Failed to fetch trades`
+        console.error(msg, e)
+        setError({ message: msg, type: 'error' })
+      } finally {
+        setIsLoading(false)
+      }
+    },
+    [],
+  )
 
   const executedSellAmount = order?.executedSellAmount.toString()
   const executedBuyAmount = order?.executedBuyAmount.toString()
   useEffect(() => {
     if (!networkId || !order?.uid) {
       return
     }
+    const controller = new AbortController()
```

Comment:
@ramirotw we could use a simple boolean variable here as well, but [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) seems more appropriate.